### ISSUE TITLE
Develop Machine Learning Model for California House Price Prediction

### ISSUE THREAD

**Issue by cute-lightfury** [cute-lightfury/California-House-Price-Predictor-Model#2]
---

### 📝 Description:

The goal is to build a regression model that predicts **California housing prices** based on features like location, number of rooms, income levels, and proximity to the ocean. We'll use the [[California Housing Dataset](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html)](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html) and train an interpretable model with good performance.

---

### 🎯 Objectives:

* Load and explore the California housing dataset
* Perform preprocessing (missing values, normalization, encoding)
* Train regression models (Linear Regression, Random Forest, Gradient Boosting)
* Evaluate using metrics (MAE, RMSE, R²)
* Export model and include prediction interface (CLI or Flask API)

---

### 💻 Tech Stack:

* **Language**: Python
* **Libraries**: `scikit-learn`, `pandas`, `matplotlib`, `xgboost`, `joblib`, `flask` (optional)

---

#### 1. Load and Explore Dataset:

```python
from sklearn.datasets import fetch_california_housing
import pandas as pd

data = fetch_california_housing()
df = pd.DataFrame(data.data, columns=data.feature_names)
df['Target'] = data.target
print(df.head())
```

#### 2. Train Model:

```python
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error

X_train, X_test, y_train, y_test = train_test_split(df.drop('Target', axis=1), df['Target'], test_size=0.2, random_state=42)

model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(X_train, y_train)
preds = model.predict(X_test)

print("RMSE:", mean_squared_error(y_test, preds, squared=False))
```

#### 3. Save and Load Model:

```python
import joblib
joblib.dump(model, 'california_model.pkl')
# later...
model = joblib.load('california_model.pkl')
```

---

### 📈 Feature Roadmap:

* [x] Dataset loading and cleaning
* [ ] Feature engineering (log transformations, income bins, ocean proximity encoding)
* [ ] Model comparison (Linear vs Tree-based models)
* [ ] Model tuning with `GridSearchCV`
* [ ] Deploy with Flask REST API or Streamlit app

---

### 🔐 Notes:

* Be sure to normalize/scale inputs if using models sensitive to feature scale.
* Include a `README.md` with setup instructions and sample input/output.
* Ensure the `.pkl` model is portable and versioned.

---

### 🔍 References:

* [[California Housing Dataset (sklearn)](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html)](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html)
* [[scikit-learn Regression Docs](https://scikit-learn.org/stable/supervised_learning.html#supervised-learning)](https://scikit-learn.org/stable/supervised_learning.html#supervised-learning)

---
